Content.Shared/Turnstile/Systems/TurnstileSystem.cs

```diff
@@ -0,0 +1,180 @@
+using Robust.Shared.Physics.Systems;
+
+namespace Content.Shared.Turnstile.Systems;
+
+public sealed class TurnstileSystem : EntitySystem
+{
+    [Dependency] private readonly SharedPhysicsSystem _physicsSystem = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearanceSystem = default!;
+    [Dependency] private readonly EntityLookupSystem _entityLookupSystem = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<TurnstileComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<TurnstileComponent, StartCollideEvent>(HandleCollide);
+        SubscribeLocalEvent<TurnstileComponent, PreventCollideEvent>(HandlePreventCollide);
+    }
+
+    private void HandlePreventCollide(Entity<TurnstileComponent> ent, ref PreventCollideEvent args)
+    {
+        // If Turnstile is rotating, allow entities the admitted entity is pulling to come through with it.
+        if (ent.Comp.State != TurnstileState.Rotating || !TryComp<PreventCollideComponent>(ent, out var preventCollide))
+            return;
+
+        if (TryComp<SharedPullerComponent>(preventCollide.Uid, out var puller))
+            args.Cancelled |= puller.Pulling == args.OtherEntity;
+    }
+
+    private void OnComponentInit(Entity<TurnstileComponent> ent, ref ComponentInit args)
+    {
+        // No turnstile should be spawned in the rotating state, as that requires a mob to be passing through.
+        if (ent.Comp.State == TurnstileState.Rotating)
+            ent.Comp.State = TurnstileState.Idle;
+
+        _appearanceSystem.SetData(ent, TurnstileVisuals.State, ent.Comp.State);
+    }
+
+
+    private void HandleCollide(Entity<TurnstileComponent> ent, ref StartCollideEvent args)
+    {
+        // The colliding entity needs to be a mob.
+        if (!HasComp<MobStateComponent>(args.OtherEntity))
+            return;
+
+        if (!TryComp<TransformComponent>(ent, out var xform))
+            return;
+
+        // Check the contact normal against our direction.
+        // For simplicity, we always want a mob to pass from the "back" to the "front" of the turnstile.
+        // That allows unanchored turnstiles to be dragged and rotated as needed, and will admit passage in the
+        // direction that they are pulled in.
+        var facingDirection = xform.LocalRotation.GetDir();
+        var directionOfContact = GetDirectionOfContact(xform, args.OtherEntity);
+
+        if (ent.Comp.State == TurnstileState.Rotating || facingDirection != directionOfContact)
+        {
+            // Reject the entity, play sound
+            _audio.PlayPredicted(ent.Comp.BumpSound, ent, args.OtherEntity, AudioParams.Default.WithVolume(-3));
+            return;
+        }
+
+        // Admit the entity.
+        var comp = EnsureComp<PreventCollideComponent>(ent);
+        comp.Uid = args.OtherEntity;
+
```

Comment:
I do not know who added this component but you should not be using it in this system.

Reply:
PreventCollideComponent usage removed. Now handles the event directly.